orchestrator/ethereum/committer/eth_committer.go

```diff
@@ -114,6 +115,34 @@ func (e *ethCommitter) SendTx(
 		return common.Hash{}, errors.Errorf("Suggested gas price %v is greater than max gas price %v", opts.GasPrice.Int64(), maxGasPrice.Int64())
 	}
 
+	// estimate gas limit
+	peggyContract := recipient
+
+	// Gas estimation cannot succeed without code for method invocations
+	code, err := e.evmProvider.PendingCodeAt(opts.Context, peggyContract)
+	if err != nil {
+		return common.Hash{}, errors.Wrap(err, "failed to get code")
+	}
+
+	if len(code) == 0 {
+		return common.Hash{}, bind.ErrNoCode
+	}
```

Comment:
why is this code needed? Seems superfluous given that we know the peggy contract is indeed a contract

Reply:
Leftover from copying [old code](https://github.com/InjectiveLabs/peggo/blob/d6bb3a187ebba2e59854100c643f8757d81b3513/orchestrator/ethereum/provider/provider.go#L79)

Removed: [0097309](https://github.com/InjectiveLabs/peggo/pull/105/commits/0097309b22b36a801f3ba98f2e558d6ad7de0f87)